tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs

```diff
@@ -481,6 +481,89 @@ public void DescriptionChanged_should_be_raised_when_the_description_changes()
             _capturedEvents.Any().Should().BeFalse();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void SelectServer_should_call_custom_selector_if_there_are_eligible_servers([Values(true, false)] bool async)
```

Comment:
Possibly change this test to assert that the custom server selector is always called whether the list is empty or not.

Reply:
I left this test as is and updated the one below to reflect new behavior.

Reply:
It could be one test that is executed more than once (a `[Theory]`) with both empty and non-empty lists.

Reply:
Done.